src/backend/distributed/utils/acquire_lock.c

```diff
@@ -211,26 +211,18 @@ LockAcquireHelperMain(Datum main_arg)
 	BackgroundWorkerInitializeConnectionByOid(args->DatabaseId, InvalidOid, 0);
 
 	/*
-	 * The query below does a self join on pg_locks to find backends that are granted a
-	 * lock our target backend (backendPid) is waiting for. Once it found such a backend
-	 * it terminates the backend with pg_terminate_pid.
+	 * The query below sends a SIGTERM signal to conflicting backends using
+	 * pg_terminate_backend() function.
 	 *
-	 * The result is are rows of pid,bool indicating backend that is terminated and the
-	 * success of the termination. These will be logged accordingly below for an
-	 * administrator to correlate in the logs with the termination message.
+	 * The result is are rows of pid,bool indicating a conflicting backend and
+	 * whether the SIGTERM was successfully delivered. These will be logged
+	 * accordingly below for an administrator to correlate in the logs with the
+	 * termination message.
 	 */
 	initStringInfo(&sql);
 	appendStringInfo(&sql,
-					 "SELECT \n"
-					 "    DISTINCT conflicting.pid,\n"
```

Comment:
As per comment on `pg_blocking_pids`

```
 * Parallel query makes it possible for there to be duplicate PIDs in the
 * result (either because multiple waiters are blocked by same PID, or
 * because multiple blockers have same group leader PID).  We do not bother
 * to eliminate such duplicates from the result.
```

Would it matter that we don't distinct the pids anymore?

Reply:
Good point, I will update the query

Reply:
@thanodnl done.
Actually, the original query had the same problem. First it was calling pg_terminate_backend() and only after that DISTINCT was applied.

Reply:
I was fearing such a thing when having a fresh look at the original query.
Thanks for fixing that here aswell!